### PR TITLE
[SYCL] Enable SPV_INTEL_fpga_dsp_control extension

### PR DESCRIPTION
**Pull request by KornevNikita**

### CODE REVIEW

**Comment by AGindinson** [comment]
This is going to conflict with https://github.com/intel/llvm/pull/3864. Please work with @MrSidims to determine the preferred order of the merges.

**Comment by MrSidims** [comment]
Both of the PRs depend on the pull-down and hence since https://github.com/intel/llvm/pull/3864 changes the test I'd prefer to merge it first.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -8636,6 +8636,7 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
         ",+SPV_INTEL_fpga_cluster_attributes,+SPV_INTEL_loop_fuse"
         ",+SPV_INTEL_long_constant_composite"
         ",+SPV_INTEL_fpga_invocation_pipelining_attributes";
+    ",+SPV_INTEL_fpga_dsp_control";
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:8639]:
```suggestion
        ",+SPV_INTEL_fpga_dsp_control";
```

Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:8639]:
```suggestion
        ",+SPV_INTEL_fpga_invocation_pipelining_attributes"
        ",+SPV_INTEL_fpga_dsp_control";
```

Comment (KornevNikita) [reply clang/lib/Driver/ToolChains/Clang.cpp:8639]: @AGindinson is causes fail in clang-format, do it anyway?

Comment (AGindinson) [reply clang/lib/Driver/ToolChains/Clang.cpp:8639]: It's just that it seems that without the change, the string is _not_ appended to `INTELExtArg`. Note the semicolon presence before the DSP control substring in the current version of the patch.

**Review by AGindinson — APPROVED** [review]

**Review by mlychkov — APPROVED** [review]

File: clang/test/Driver/sycl-spirv-ext.c [hunk 2]
```diff
@@ -47,6 +47,7 @@
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_cluster_attributes,+SPV_INTEL_loop_fuse
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_long_constant_composite
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes"
+// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_dsp_control
```
Comment (MrSidims) [inline clang/test/Driver/sycl-spirv-ext.c:50]:
```suggestion
// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes
// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_dsp_control"
```

File: clang/test/Driver/sycl-spirv-ext.c [hunk 3]
```diff
@@ -70,3 +71,4 @@
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_long_constant_composite
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_usm_storage_classes"
+// CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_dsp_control
```
Comment (MrSidims) [inline clang/test/Driver/sycl-spirv-ext.c:74]:
```suggestion
// CHECK-FPGA-HW-SAME:,+SPV_INTEL_usm_storage_classes
// CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_dsp_control"
```

**Review by MrSidims — APPROVED** [review]

**Review by AGindinson — DISMISSED** [review]

File: clang/test/Driver/sycl-spirv-ext.c [hunk 4]
```diff
@@ -46,7 +46,8 @@
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_variable_length_array,+SPV_INTEL_fp_fast_math_mode
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_cluster_attributes,+SPV_INTEL_loop_fuse
 // CHECK-DEFAULT-SAME:,+SPV_INTEL_long_constant_composite
-// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes"
+// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes
+// CHECK-DEFAULT-SAME:,+SPV_INTEL_fpga_dsp_control"
```
Comment (mlychkov) [inline clang/test/Driver/sycl-spirv-ext.c:50]: I'd even removed quotation mark from check string.

Comment (mlychkov) [reply clang/test/Driver/sycl-spirv-ext.c:50]: On the other hand, it checks that no other extensions are added unexpectedly. So, nevermind.

**Review by mlychkov — APPROVED** [review]

**Review by bader — CHANGES_REQUESTED** [review]
Please, fix pre-commit test failures.

File: clang/test/Driver/sycl-spirv-ext.c [hunk 5]
```diff
@@ -69,4 +70,5 @@
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_cluster_attributes,+SPV_INTEL_loop_fuse
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_long_constant_composite
 // CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_invocation_pipelining_attributes
-// CHECK-FPGA-HW-SAME:,+SPV_INTEL_usm_storage_classes"
+// CHECK-FPGA-HW-SAME:,+SPV_INTEL_usm_storage_classes
+// CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_dsp_control"
```
Comment (mlychkov) [inline clang/test/Driver/sycl-spirv-ext.c:74]:
```suggestion
// CHECK-FPGA-HW-SAME:,+SPV_INTEL_fpga_dsp_control
// CHECK-FPGA-HW-SAME:,+SPV_INTEL_usm_storage_classes"
```

**Review by AGindinson — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**